app/controllers/user_filters_controller.rb

```diff
@@ -0,0 +1,47 @@
+# frozen_string_literal: true
+
+# UserFiltersController: used to manage user saved advanced filters
+class UserFiltersController < ApplicationController
+  before_action :authenticate_user!
+  before_action :check_role
+
+  def index
+    render json: current_user.user_filters
+  end
+
+  def create
+    redirect_to root_url unless current_user.user_filters.count < 100 # Enforce upper limit per user
```

Comment:
I agree, but couldn't really decide on what that limit should even be. I'd like to let people add as many as they'd like, since this is cheap resource wise, but wanted some sort of blocker for the 0.000001% of people who might try to abuse this.